### PR TITLE
feat: [CI-14420]: FileOps migration: FileCopyOperation support

### PR DESCRIPTION
**Pull request by vanisrikanithi**
Added "FileCopyOperation " support to go-convert specific to jenkins migration.

FileCopy Jenkins Trace: [jfileops__filecopy.json]
[fileops__filecopy.json](https://github.com/user-attachments/files/17140943/fileops__filecopy.json)

go run main.go jenkinsjson --downgrade ~/go/fileops__filecopy.json
```
pipeline:
  identifier: default
  name: default
  orgIdentifier: default
  projectIdentifier: default
  properties:
    ci:
      codebase:
        build: <+input>
  stages:
  - stage:
      identifier: build
      name: build
      spec:
        cloneCodebase: true
        execution:
          steps:
          - stepGroup:
              identifier: Stage__null4661c3
              name: Prepare Workspace
              steps:
              - step:
                  identifier: bat2cddaa
                  name: bat_bat
                  spec:
                    command: |-
                      echo @"
                      if not exist src mkdir src
                      "@ > bat2cddaa.bat
                       ./bat2cddaa.bat
                      echo @"
                      if not exist dest mkdir dest
                      "@ > bat42c603.bat
                       ./bat42c603.bat
                    shell: Pwsh
                  timeout: ""
                  type: Run
              timeout: ""
          - stepGroup:
              identifier: Stage__nullaa3dc7
              name: Create Files
              steps:
              - step:
                  identifier: bat6c99e8
                  name: bat_bat
                  spec:
                    command: |-
                      echo @"
                      echo Sample text 1 > src\sample1.txt
                      "@ > bat6c99e8.bat
                       ./bat6c99e8.bat
                      echo @"
                      echo Sample text 2 > src\sample2.txt
                      "@ > bat6b80e2.bat
                       ./bat6b80e2.bat
                    shell: Pwsh
                  timeout: ""
                  type: Run
              timeout: ""
          - stepGroup:
              identifier: Stage__null583b84
              name: Copy Files
              steps:
              - step:
                  identifier: fileoperationsb888f2
                  name: fileCopyOperation
                  spec:
                    command: cp -r src/*.txt dest/
                    image: alpine
                  timeout: ""
                  type: Run
              timeout: ""
        platform:
          arch: Amd64
          os: Linux
        runtime:
          spec: {}
          type: Cloud
      type: CI
```

### CODE REVIEW

File: convert/jenkinsjson/convert.go [hunk 1]
```diff
@@ -801,12 +848,21 @@ func mergeRunSteps(steps *[]StepWithID) {
 			merged = append(merged, cursor)
 			cursor = current
 			pushed = true
+
+			if len(*steps) == i+1 {
+				fmt.Println("Inside if line 875")
```
Comment (jamie-harness) [inline convert/jenkinsjson/convert.go:853]: remove this please

File: convert/jenkinsjson/convert.go [hunk 2]
```diff
@@ -801,12 +848,21 @@ func mergeRunSteps(steps *[]StepWithID) {
...
+				isLastStep = true
+			}
 		}
 	}
 
 	if !pushed {
 		merged = append(merged, cursor)
 	}
+
+	if pushed && isLastStep {
```
Comment (jamie-harness) [inline convert/jenkinsjson/convert.go:863]: please revert this change and sync with latest. It should have been fixed already. If not please contact me with the example you are trying

**Comment by Ompragash** [comment]
merged in #225
